sdk/log: fix "limit reached" logging and Record.DroppedAttributes (#7662)
open-telemetry/opentelemetry-go: CHANGELOG.md

## [Unreleased]

### Fixed

- Fix bad log message when key-value pairs are dropped because of key duplication in `go.opentelemetry.io/otel/sdk/log`. (#7662)
- Fix `DroppedAttributes` in `go.opentelemetry.io/otel/sdk/log` to not count the non-attribute key-value pairs dropped because of key duplication. (#7662)

> - Fix `setDropped` in `go.opentelemetry.io/otel/sdk/log` to log dropped keys only when the input quantity is greater than zero. (#7662)

Reviewer: ```suggestion
- Fix `SetAttributes` in `go.opentelemetry.io/otel/sdk/log` to not log that log attributes are dropped when they are actually not. (#7662)

```
